metabase/metabase
# fix(View Details): calculate row after vertical and horizontal scroll

Closes https://github.com/metabase/metabase/issues/51301

### Description

This problem appears when you first scroll vertically then horizontally, `ScrollSync` from `react-virtualized` triggers 2 scroll events (one for Header Grid and second for the main Grid) and one of events puts scrollTop (most likely header) as `undefined` since there is no way to scroll it. Because of that, consequence of vertical and horizontal scroll led to the wrong calculation of view details icon.

### How to verify

0. make browser to take half of the screen
1. New question -> feedback table -> visualize
2. scroll vertically by half a row, then horizontally, put a mouse over some line - view details view icon should appear
3. click on details view icon -> wrong line's data is shown in the modal

### Demo

### Before

https://github.com/user-attachments/assets/a16a316f-179b-4eff-aad1-27efa37d5fea

<img width="1284" alt="Screenshot 2024-12-14 at 00 14 18" src="https://github.com/user-attachments/assets/5b022ef5-dd2e-4fb4-94a6-0f549c310f16" />


### After


<img width="1282" alt="Screenshot 2024-12-14 at 00 14 37" src="https://github.com/user-attachments/assets/41302ac6-5ef4-4fc6-8f96-73a980bc3ec1" />




### Checklist

- [x] [stress test](https://github.com/metabase/metabase/actions/runs/12349334779/job/34459960701)
